pkg/kubelet/network/exec/exec.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exec
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network"
+	cmdexec "github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/golang/glog"
+)
+
+type execNetworkPlugin struct {
+	execName string
+	execPath string
+	host     network.Host
+}
+
+const (
+	initCmd     = "--init"
+	setUpCmd    = "--setup"
+	tearDownCmd = "--teardown"
+	execDir     = "/usr/libexec/kubernetes/kubelet-plugins/net/exec/"
+)
+
+func ProbeNetworkPlugins() []network.NetworkPlugin {
+	return ProbeNetworkPluginsWithExecDir(execDir)
+}
+
+func ProbeNetworkPluginsWithExecDir(pluginDir string) []network.NetworkPlugin {
+	execPlugins := []network.NetworkPlugin{}
+
+	files, _ := ioutil.ReadDir(pluginDir)
+	for _, f := range files {
+		// only directories are counted as plugins
+		// and we expect pluginDir/dirname/dirname to be an executable
+		if f.IsDir() {
+			execPath := path.Join(pluginDir, f.Name())
+			execPlugins = append(execPlugins, &execNetworkPlugin{execName: f.Name(), execPath: execPath})
+		}
+	}
+	return execPlugins
+}
+
+func (plugin *execNetworkPlugin) Init(host network.Host) error {
+	plugin.host = host
+	// call the init script
+	executable := path.Join(plugin.execPath, plugin.execName)
+	out, err := cmdexec.New().Command(executable, initCmd).CombinedOutput()
+	glog.V(5).Infof("Init 'exec' network plugin output: %s, %v", string(out), err)
+	return err
+}
+
+func (plugin *execNetworkPlugin) Name() string {
+	return plugin.execName
+}
+
+func (plugin *execNetworkPlugin) Validate() error {
+	executable := path.Join(plugin.execPath, plugin.execName)
+	fi, err := os.Stat(executable)
+	if err != nil {
+		return err
+	}
+	if fi.Mode()&0111 == 0 {
```

Comment:
Isn't there a constant for 0111?  Do we really need 0111?  what I think you want is `syscall.Access(path, 0x1)`.  See https://github.com/golang/go/issues/6262 for why 0x1 is needed - golang just hasn't defined a constant for it yet (2 years after the bug was filed, sigh)